### PR TITLE
🧪 Experiment: story auto advance and new cta button

### PR DESCRIPTION
**Pull request by powerivq**
Experiments:

1. Control
2. Auto advance + old CTA
3. Auto advance + new CTA w/ animation
4. Auto advance + new CTA w/o animation

Experiment IDs need to be updated before merging.

### CODE REVIEW

File: src/experiments/story-ad-progress-segment.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/**
+ * @const
+ * @type {Object<string, string>}
+ */
+export const StoryAdSegmentExp = {
+  ID: 'story-ad-auto-advance',
+  CONTROL: 't', // No change
+  AUTO_ADVANCE_OLD_CTA: 'a', // Auto advance, old CTA button
+  AUTO_ADVANCE_NEW_CTA: 'b', // Auto advance, new CTA button
+  AUTO_ADVANCE_NEW_CTA_NOT_ANIMATED: 'c', // Auto advance, new CTA button but not animated
```
Comment (calebcordry) [inline src/experiments/story-ad-progress-segment.js:10]: As discussed offline lets make sure we fill in these ids before submitting.

File: extensions/amp-story/1.0/amp-story-page.js [hunk 2]
```diff
@@ -1105,8 +1106,14 @@ export class AmpStoryPage extends AMP.BaseElement {
   emitProgress_(progress) {
     // Don't emit progress for ads, since the progress bar is hidden.
     // Don't emit progress for inactive pages, because race conditions.
+    const storyAdSegmentBranch = getExperimentBranch(
+      this.win,
+      StoryAdSegmentExp.ID
+    );
     if (
-      (!isExperimentOn(this.win, 'story-ad-auto-advance') && this.isAd()) ||
+      ((!storyAdSegmentBranch ||
```
Comment (calebcordry) [inline extensions/amp-story/1.0/amp-story-page.js:1114]: nit: these complex if statements here and below are a bit hard to parse, maybe pulling to a named var will help.

**Review by calebcordry — COMMENTED** [review]
Experiment setup LGTM. Will defer to the other folks on the best way to disable animation

**Review by calebcordry — APPROVED** [review]

File: extensions/amp-story-page-attachment/0.1/amp-story-open-page-attachment.css [hunk 3]
```diff
@@ -168,6 +168,10 @@
   animation: i-amphtml-tap-scale var(--i-amphtml-page-attachment-ui-animation-duration) var(--i-amphtml-page-attachment-ui-animation-delay) both !important;
 }
 
+.i-amphtml-story-page-open-attachment-outlink-no-animation-exp.i-amphtml-story-page-open-attachment-outlink-no-animation-exp[active] {
```
Comment (mszylkowski) [inline extensions/amp-story-page-attachment/0.1/amp-story-open-page-attachment.css:171]: If you want to avoid the animations above, it's better to do `.i-amphtml-story-page-open-attachment-outlink[active]:not(.i-amphtml-story-page-open-attachment-outlink-no-animation-exp) {animation: i-amphtml-tap-scale...}` instead of overriding it with `animation: none` on a new property. Same below.

File: extensions/amp-story-page-attachment/0.1/amp-story-open-page-attachment.js [hunk 4]
```diff
@@ -128,13 +132,21 @@ const renderOutlinkUI = (pageEl, attachmentEl) => {
   // Set image.
   const openImgAttr = attachmentEl.getAttribute('cta-image');
 
+  const noAnimation =
+    getExperimentBranch(getWin(pageEl), 'story-ad-auto-advance') == 'c';
```
Comment (mszylkowski) [inline extensions/amp-story-page-attachment/0.1/amp-story-open-page-attachment.js:136]: Where is this constant `c` from? Do you need to replace it with the correct value from `StoryAdSegmentExp.AUTO_ADVANCE_NEW_CTA_NOT_ANIMATED`?

File: extensions/amp-story/1.0/progress-bar.js [hunk 5]
```diff
@@ -155,12 +157,17 @@ export class ProgressBar {
...
+            this.win_,
+            StoryAdSegmentExp.ID
+          );
           /** @type {!Array} */ (pageIds).forEach((id) => {
             if (
               // Do not show progress bar for the ad page (control group).
               // Show progress bar for the ad page (experiment group).
               (!this.isAdSegment_(id) ||
-                isExperimentOn(this.win_, 'story-ad-auto-advance')) &&
+                (storyAdSegmentBranch &&
```
Comment (mszylkowski) [inline extensions/amp-story/1.0/progress-bar.js:169]: If `storyAdSegmentBranch` is null, you don't need to check `x && x == CONTROL`, you can just check `x == CONTROL` since `CONTROL` is not null.  You can just check `storyAdSegmentBranch != StoryAdSegmentExp.CONTROL`.

Comment (mszylkowski) [reply extensions/amp-story/1.0/progress-bar.js:169]: Nvm

**Review by mszylkowski — APPROVED** [review]
